sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -396,6 +396,26 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
 object SparkPlan {
   private[execution] val subqueryExecutionContext = ExecutionContext.fromExecutorService(
     ThreadUtils.newDaemonCachedThreadPool("subquery", 16))
+
+  /**
+   * Returns if the actual ordering satisfies the required ordering.
+   *
+   * Ordering A satisfies ordering B if and only if B is an equivalent of A or of A's prefix.
+   */
+  def orderingSatisfies(actual: Seq[SortOrder], required: Seq[SortOrder]): Boolean = {
```

Comment:
how about moving this to `SortOrder` object : https://github.com/apache/spark/blob/e9c91badce64731ffd3e53cbcd9f044a7593e6b8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala#L92

Reply:
Actually I had struggled where to put this, in SortOrder or SparkPlan. It doesn't look like anywhere else is using Seq[SortOrder] so far, so I chose to leave this out of SortOrder. I think, though, if we see potential usage of Seq[SortOrder] elsewhere, it might be worth to wrap it as a class. Agree? Either way, I could put it into SortOrder for now.

Reply:
SparkPlan is the node for physical operator in SQL so doesn't feel like a good place to have this. Since one would want to have all methods related to `SortOrder` in a single place, the object class feels better option. We can revisit if there are more such methods being added to that object and refac to a class.

Reply:
Let us first move it to `SortOrder`.